### PR TITLE
Fix wrong assumption in routing socket mbuf copy.

### PR DESCRIPTION
**Pull request by qwattash**
The previous assumption of existing mbuf trailing space is wrong, instead we
use a slower path via m_copyback that should always work.

### CODE REVIEW

File: sys/net/rtsock.c [hunk 1]
```diff
@@ -1255,21 +1256,15 @@ rtsock_msg_mbuf(int type, struct rt_addrinfo *rtinfo)
...
-			 * XXX-AM: This is a bit gross but avoids calling
-			 * m_copyback or m_append to copy zeroes.
-			 */
-			KASSERT(M_TRAILINGSPACE(m) >= dlen - sa->sa_len,
-			    ("No space for padding in rtsock mbuf"));
-			bzero(mtod(m, caddr_t) + len + sa->sa_len,
-			    dlen - sa->sa_len);
-			if (m->m_flags & M_PKTHDR)
-				m->m_pkthdr.len += dlen - sa->sa_len;
+			KASSERT(dlen - sa->sa_len <= sizeof(long),
```
Comment (arichardson) [inline sys/net/rtsock.c:1264]:
```suggestion
			char zero_padding[sizeof(long)] = {0};

			KASSERT(dlen - sa->sa_len <= sizeof(long),
```
A zero-filled char array would avoid the need for a cast. Don't really mind either way.

**Review by arichardson — APPROVED** [review]
I can confirm that this fixes the panic I was seeing.

File: sys/net/rtsock.c [hunk 2]
```diff
@@ -1255,21 +1257,15 @@ rtsock_msg_mbuf(int type, struct rt_addrinfo *rtinfo)
...
+			    ("Too much padding required"));
+			m_copyback(m, len + sa->sa_len, dlen - sa->sa_len,
+			    &zero_padding);
```
Comment (arichardson) [inline sys/net/rtsock.c:1268]:
```suggestion
			    zero_padding);
```

File: sys/net/rtsock.c [hunk 3]
```diff
@@ -1203,6 +1203,8 @@ rtsock_msg_mbuf(int type, struct rt_addrinfo *rtinfo)
 	struct sockaddr_storage ss;
 	struct sockaddr_in6 *sin6;
 #endif
+	long padding = 0;
```
Comment (arichardson) [inline sys/net/rtsock.c:1206]:
```suggestion
```

Comment (qwattash) [reply sys/net/rtsock.c:1206]: Thought I deleted it :(

**Review by arichardson — APPROVED** [review]

**Pull request merged**